pkgs/applications/science/misc/golly/default.nix

```
@@ -35,24 +36,45 @@ stdenv.mkDerivation rec {
     wrapGAppsHook3
   ];
 
+  enableParallelBuilding = true;
+
+  preConfigure = lib.optionalString stdenv.isDarwin ''
+    export NIX_CFLAGS_COMPILE+=" -L$SDKROOT/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries"
+  '';
+
   setSourceRoot = ''
     sourceRoot=$(echo */gui-wx)
   '';
 
-  postPatch = ''
-    sed -e 's@PYTHON_SHLIB@${python3}/lib/libpython3.so@' -i wxprefs.cpp
-    ! grep _SHLIB *.cpp
+  postPatch =
+    ''
+      sed -e 's@PYTHON_SHLIB@${python3}/lib/libpython3.so@' -i wxprefs.cpp
 
-    grep /lib/libpython wxprefs.cpp
-  '';
+      # Use CXX from environment, falling back to g++ if not set.
+      sed -i "s|^CXXC = g++|CXXC = \$(CXX)|g" makefile-gtk
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      # Remove --as-needed linker flag on Darwin, it's not supported
+      sed -i "s|-Wl,--as-needed||g" makefile-gtk
+    '';
 
   makeFlags = [
     "-f"
     "makefile-gtk"
     "ENABLE_SOUND=1"
     "GOLLYDIR=${placeholder "out"}/share/golly"
+    "CC=${stdenv.cc.targetPrefix}cc"
+    "CXX=${stdenv.cc.targetPrefix}c++"
+    "LD=${stdenv.cc.targetPrefix}c++"
   ];
 
+  buildPhase = ''
+    runHook preBuild
+    (cd ../lua && make CC="${stdenv.cc.targetPrefix}cc" all)
+    make $makeFlags
+    runHook postBuild
+  '';
```

```suggestion
  preBuild = ''
    (cd ../lua && make CC="${stdenv.cc.targetPrefix}cc" all)
  '';
```
Should be equivalent